plugins/woocommerce/src/Internal/ProductDownloads/ApprovedDirectories/Register.php

```diff
@@ -240,13 +243,20 @@ public function get_by_url( string $url ) {
 	public function is_valid_path( string $download_url ): bool {
 		global $wpdb;
 
+		if ( $download_url && array_key_exists( $download_url, $this->valid_paths ) ) {
+			return $this->valid_paths[ $download_url ];
+		}
+
 		$parent_directories = array();
 
 		foreach ( ( new URL( $this->normalize_url( $download_url ) ) )->get_all_parent_urls() as $parent ) {
 			$parent_directories[] = "'" . esc_sql( $parent ) . "'";
 		}
 
 		if ( empty( $parent_directories ) ) {
+
+			$this->valid_paths[ $download_url ] = false;
+
```

Comment:
```suggestion
			$this->valid_paths[ $download_url ] = false;

```
Removing extra line at the top...